crates/telio-relay/src/derp/mod.rs

```diff
@@ -565,11 +565,18 @@ impl Runtime for State {
 
         match &mut self.conn {
             Some(c) => {
+                let (upper_read, chan_tx) = match self.channel.as_mut() {
+                    Some(c) => (c.rx.recv(), &c.tx),
+                    None => {
+                        // Similar case to config, but the log is already printed
+                        return (update.await)(self).await;
+                    }
+                };
```

Comment:
```suggestion
               if self.channel.is_none() {
                    return (update.await)(self).await;
                }
```

isn't this enough?

Reply:
It could be replaced by a flag and the option from channel could be removed, or probably in some other way as well, I thought that it is more clear this way, but I'm not really attached to it, so if you would like some other solution more I can change it.

Reply:
the channel was still defined later, but it's ok this is nitpicking